ModelVale/ModelVale/ViewControllers/Main/DataViewController.m

```diff
@@ -16,57 +16,67 @@
 #import "AddDataViewController.h"
 
 @interface DataViewController () <UICollectionViewDelegate, UICollectionViewDataSource>
+@property (weak, nonatomic) IBOutlet UIButton *loadMoreButton;
 @property (weak, nonatomic) IBOutlet UICollectionView *userDataCollectionView;
 @end
 
 @implementation DataViewController
 
-// Need two section headers (or rather, testTrainTypeArray.count from TestTrainEnum.h) for each label
+// Make two section headers (or rather, testTrainTypeArray.count from TestTrainEnum.h) for each label
 - (void)viewDidLoad {
     [super viewDidLoad];
-    //XXX todo remove modelLabel
-//    self.modelLabels = [NSMutableArray new];
+    self.modelLabels = [NSMutableArray new];
     self.userDataCollectionView.delegate = self;
     self.userDataCollectionView.dataSource = self;
-    
-    NSMutableArray* modelLabels = self.model.labeledData;
+    self.labelFetchStart = 0;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    [self fetchLocalData:^{
+        [self.userDataCollectionView reloadData];
```

Comment:
Since I assume that fetchLocalData will be an async call and can happen on another thread than the main thread I will suggest that you dispatch the reloadData call to the main queue. 

dispatch_async(dispatch_get_main_queue(), ^{
   // do work here
});